### ISSUE TITLE
Allow opening devtools programmatically

### ISSUE THREAD

**Issue by lucasfernog** [tauri-apps/wry#287]
See https://github.com/tauri-apps/tauri/issues/1213.

On Linux it can be done with https://docs.rs/webkit2gtk/0.11.0/webkit2gtk/trait.WebInspectorExt.html#tymethod.show

**Comment by lemarier** [comment]
I got it to works on windows and linux. 

Mac is a pain i need to keep trying… i’ll update my progress here.

**Comment by amrbashir** [comment]
I think last time we checked macOs, it was a private api

**Comment by wusyong** [comment]
yes it's a private api on mac. I don't feel like expose a method is a good idea in this case.
What do you think about setting some env variable so it could only affect debug build.

**Comment by lucasfernog** [comment]
I think we can expose a method and hide it under `#[cfg(debug_assertions)]`

**Comment by wusyong** [comment]
Here's the method on Windows: https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2?view=webview2-1.0.664.37#opendevtoolswindow

**Comment by schickling** [comment]
Thanks a lot for pushing this over the finish line @wusyong. Is this already somewhere in the docs?

**Comment by FabianLars** [comment]
@schickling if you're asking about wry then yes:
https://docs.rs/wry/latest/wry/webview/struct.WebView.html#method.devtool
https://docs.rs/wry/latest/wry/webview/struct.WebViewBuilder.html#method.with_dev_tool
You need to enable the devtool feature flag tho: https://docs.rs/wry/latest/wry/index.html#feature-flags
Dev tools used to be enabled on debug builds only, this got replaced with this flag (so the flag is not only needed for the functions but also the dev tools in general)

If you're asking about tauri, then the docs won't be available until the next release (really really soon). If you're using the git version of tauri you can see the docs in the PR code: https://github.com/tauri-apps/tauri/pull/3350/files (this is not merged yet!).
tl;dr: enabled on debug builds and with the same devtool flag, so you don't need to build tauri (or wry) in debug mode anymore to ship your app with enabled devtools (mind the private API usage on macOS)

**Comment by KaKi87** [comment]
> the next release (really really soon)

Could we be notified at that time ? Thanks

**Comment by FabianLars** [comment]
> Could we be notified at that time?

We have a discord channel for that and you can subscribe to GitHub notifications (watch -> releases only).

Otherwise I'd have to spend a whole day manually pinging people haha (you're not the first one askin for a notification, that's why)

**Comment by KaKi87** [comment]
> We have a discord channel for that

Could a role mention be added to it ?

> you can subscribe to GitHub notifications

Yeah but those are email-based.

Thanks

**Comment by FabianLars** [comment]
We will most likely do a `@here` ping in either the news or the releases channel (probably both).
